oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java

```diff
@@ -376,15 +374,63 @@ public SampleFamily endpoint(List<String> serviceKeys, List<String> endpointKeys
         if (this == EMPTY) {
             return EMPTY;
         }
-        this.context.setMeterEntity(MeterEntity.newEndpoint(dim(serviceKeys), dim(endpointKeys)));
-        return left(io.vavr.collection.Stream.concat(serviceKeys, endpointKeys).asJava());
+        return createMeterSamples(serviceKeys, endpointKeys, ScopeType.ENDPOINT);
+    }
+
+    private SampleFamily createMeterSamples(List<String> serviceKeys, List<String> level2keys, ScopeType scopeType) {
```

Comment:
I think we need an inherent structure rather than this 3 parameter. Such as `interface EntityDescription` with different implementations.

Reply:
Basically, I prefer to remove the name `level2*`. This doesn't exist in the SkyWalking, I think.

Reply:
Done